osu.Game/Overlays/Profile/ProfileHeader.cs

```diff
@@ -2,54 +2,111 @@
 
         public readonly SupporterIcon SupporterTag;
         private readonly Container coverContainer;
-        private readonly Sprite levelBadge;
-        private readonly SpriteText levelText;
-        private readonly GradeBadge gradeSSPlus, gradeSS, gradeSPlus, gradeS, gradeA;
-        private readonly Box colourBar;
-        private readonly DrawableFlag countryFlag;
-        private readonly BadgeContainer badgeContainer;
-
-        private const float cover_height = 350;
-        private const float info_height = 150;
-        private const float info_width = 220;
+        private readonly OsuSpriteText coverInfoText;
+        private readonly ProfileHeaderTabControl infoTabControl;
+
+        private readonly Box headerTopBox;
+        private readonly UpdateableAvatar avatar;
+        private readonly OsuSpriteText usernameText;
+        private readonly ExternalLinkButton openUserExternally;
+        private readonly OsuSpriteText titleText;
+        private readonly DrawableFlag userFlag;
+        private readonly OsuSpriteText userCountryText;
+        private readonly Box userIconSeperatorBox;
+        private readonly FillFlowContainer userStats;
+
+        private readonly Box headerCenterBox;
+        private readonly OsuSpriteText followerText;
+        private readonly ProfileHeaderButton messageButton;
+        private readonly ProfileHeaderButton expandButton;
+        private readonly Sprite levelBadgeSprite;
+        private readonly OsuSpriteText levelBadgeText;
+
+        private readonly Bar levelProgressBar;
+        private readonly OsuSpriteText levelProgressText;
+
+        private readonly OverlinedInfoContainer hiddenDetailGlobal, hiddenDetailCountry;
+
+        public readonly BindableBool DetailsVisible = new BindableBool();
+
+        private readonly Box headerDetailBox;
+        private readonly HasTooltipContainer totalPlayTimeTooltip;
+        private readonly OverlinedInfoContainer totalPlayTimeInfo, medalInfo, ppInfo;
+        private readonly Dictionary<ScoreRank, ScoreRankInfo> scoreRankInfos = new Dictionary<ScoreRank, ScoreRankInfo>();
+        private readonly OverlinedInfoContainer detailGlobalRank, detailCountryRank;
+
+        private readonly Box headerBadgeBox;
+        private readonly FillFlowContainer badgeFlowContainer;
+        private readonly Container badgeContainer;
+
+        private readonly Box headerBottomBox;
+        private readonly LinkFlowContainer bottomTopLinkContainer;
+        private readonly LinkFlowContainer bottomLinkContainer;
+
+        private const float cover_height = 150;
+        private const float cover_info_height = 75;
         private const float avatar_size = 110;
-        private const float level_position = 30;
-        private const float level_height = 60;
-        private const float stats_width = 280;
+        private static readonly Dictionary<string, string> play_styles = new Dictionary<string, string>
```

Comment:
make this an enum instead, using `[Description("Touch Screen")]`

Reply:
I'd have to manually convert the json string into the enum though...